### PR TITLE
roxanaluca-assignment4

### PR DESCRIPTION
**Pull request by roxanaluca**

### CODE REVIEW

File: roxanaluca/assignment-4/Main/src/main/Main.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+       char[][] map = {{'S','L','S','L'},
+                       {'L','L','S','S'},
+                       {'S','S','L','S'},
+                       {'S','S','L','S'}};
+       System.out.println(countIslands(map));
+    }
+    
+    public static int countIslands(char[][] map)
+    {
+        int nr = 0;int[][] count = new int[map.length][map[0].length];
```
Comment (imorlowska) [inline roxanaluca/assignment-4/Main/src/main/Main.java]: nit: according to the Google jaca style guide please don't do multiple definitions in one line

File: roxanaluca/assignment-4/Main/src/main/Main.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        
+        for (int i = 0;i<map.length;i++)
+            for (int j = 0;j<map[0].length;j++)
+                if (map[i][j] =='L' && count[i][j] == 0)
```
Comment (imorlowska) [inline roxanaluca/assignment-4/Main/src/main/Main.java]: You're using an array of integers to keep track which islands they belong to - we only care about how many there are, so we could have an array of Boolean values instead to indicate if a field was visited or not

File: roxanaluca/assignment-4/Main/test/Tests.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+/**
+ *
+ * @author mariaroxanaluca
+ */
+public class Tests {
```
Comment (imorlowska) [inline roxanaluca/assignment-4/Main/test/Tests.java]: Try thinking of edge cases - for example L shaped islands or O shaped island with another island in the middle etc. The increase in the size of a map is a good thing to check to have, but it doesn't cover edge cases. What if you had an empty array?

File: roxanaluca/assignment-4/Main/src/main/Main.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+       //first test
+       
+       char[][] map = {{'S','L','S','L'},
+                       {'L','L','S','S'},
+                       {'S','S','L','S'},
+                       {'S','S','L','S'}};
+       System.out.println(countIslands(map));
+    }
+    
+    public static int countIslands(char[][] map)
```
Comment (imorlowska) [inline roxanaluca/assignment-4/Main/src/main/Main.java]: Please add javadoc (class level comment) explaining what the method does, what are the inputs, what is the format and what is the expected return. Also please try analysing the complexity of the solution and include that in the comment.

**Comment by imorlowska** [comment]
Please iterate on the solution - I pointed out a few things in the comments, please address them.
